packages/react-utilities/src/compose/types.ts

```diff
@@ -76,6 +71,6 @@ export type ComponentState<
   ShorthandProps extends keyof Props = never,
   DefaultedProps extends keyof ResolvedShorthandProps<Props, ShorthandProps> = never
 > = RequiredProps<ResolvedShorthandProps<Props, ShorthandProps>, DefaultedProps> & {
-  as?: React.ElementType;
+  as?: keyof JSX.IntrinsicElements;
```

Comment:
I don't recall why `as` is duplicated here, but I don't think it should be necessary (and is not desirable), given that it exists on `ComponentProps`. Does anything break if you remove it here?